test/test_jit.py

```diff
@@ -3544,6 +3544,33 @@ def bar(x):
         x = torch.rand(3, 4)
         self.assertEqual(bar(x), (x * x + 3 * x) * (x * x + 3 * x))
 
+    def test_static_methods(self):
+        class M(nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+
+            @staticmethod
+            def my_method(x):
+                return x + 100
+
+            def forward(self, x):
+                return x + M.my_method(x)
+
+        class N(nn.Module):
+            def __init__(self):
+                super(N, self).__init__()
+
+            @staticmethod
+            def my_method(x):
+                return x * 100
+
+            def forward(self, x):
+                return x - M.my_method(x) + N.my_method(x)
```

Comment:
How does this serialize if we're not inlining methods ? Could you print out the source ?

Reply:
We still do inlining:

```
graph(%self : ClassType<N>,
      %x.1 : Tensor):
  %23 : int = prim::Constant[value=100]() # test/test_jit.py:3654:27
  %22 : int = prim::Constant[value=1]()
  %24 : Tensor = aten::add(%x.1, %23, %22) # test/test_jit.py:3654:23
  %7 : Tensor = aten::sub(%x.1, %24, %22) # test/test_jit.py:3668:23
  %26 : Tensor = aten::mul(%x.1, %23) # test/test_jit.py:3665:23
  %12 : Tensor = aten::add(%7, %26, %22) # test/test_jit.py:3668:23
  return (%12)
```

With inlining off the methods get correctly mangled:

```
graph(%self : ClassType<N>,
      %x.1 : Tensor):
  %9 : Function = prim::Constant[name="my_method"]()
  %6 : int = prim::Constant[value=1]()
  %4 : Function = prim::Constant[name="my_method"]()
  %5 : Tensor = prim::CallFunction(%4, %x.1) # test/test_jit.py:3668:27
  %7 : Tensor = aten::sub(%x.1, %5, %6) # test/test_jit.py:3668:23
  %10 : Tensor = prim::CallFunction(%9, %x.1) # test/test_jit.py:3668:44
  %12 : Tensor = aten::add(%7, %10, %6) # test/test_jit.py:3668:23
  return (%12)
```

```
Archive:  a.pt
 extracting: a/version               
 extracting: a/data.pkl              
  inflating: a/code/__torch__.py     
  inflating: a/code/__torch__.py.debug_pkl  
  inflating: a/code/__torch__/___torch_mangle_0.py  
  inflating: a/code/__torch__/___torch_mangle_0.py.debug_pkl  
  inflating: a/code/__torch__/___torch_mangle_1.py  
  inflating: a/code/__torch__/___torch_mangle_1.py.debug_pkl  
 extracting: a/constants.pkl         
[dev] ~/d/pytorch > cat ^C
[dev] ~/d/pytorch > cd ^C
[dev] ~/d/pytorch > cat a/code/__torch__/___torch_mangle_0.py   (base) [ 15:22:38 ]
op_version_set = 1
def my_method(x: Tensor) -> Tensor:
  return torch.add(x, 100, 1)
[dev] ~/d/pytorch > cat a/code/__torch__/___torch_mangle_1.py   (base) [ 15:22:50 ]
op_version_set = 1
def my_method(x: Tensor) -> Tensor:
  return torch.mul(x, 100)
[dev] ~/d/pytorch >                                             (base) [ 15:22:53 ]
```

and the tests still work